### PR TITLE
Feature/#58

### LINKED ISSUE #58: Form labels should focus the form control / SPA-1265
In angular we don't use the class element IDs and therefor we can't do:
{code:java}
<label for="FORM_CONTROL_ID">label text</label>
<input id="FORM_CONTROL_ID">{code}
Instead, we should ideally do:
{code:java}
<label>
  label text
  <input>
</label>
{code}
This is both a fix as well as a simplification. 

This should be fixed in the various forms we have (login, address forms, checkout, etc.)

### PR DESCRIPTION
**Pull request by Matejk00**
Fix #58

### CODE REVIEW

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/payment-method/payment-form/payment-form.component.html [hunk 1]
```diff
@@ -6,50 +6,67 @@
                 <div class="form-group">
                     <ng-container *ngIf="cardTypes$ | async as cardTypes">
                         <div *ngIf="cardTypes.length !== 0">
-                            <label for="CardType" aria-required="true">Payment Type</label>
-                            <ng-select [searchable]="false" [clearable]="false" [items]="cardTypes" bindLabel="name"
-                                bindValue="code" placeholder="Select One..." (change)="paymentSelected($event)">
-                            </ng-select>
+                            <label>
+                                <span class="label-content required" aria-required="true">
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/payment-method/payment-form/payment-form.component.html]: Does `aria-required` work on span?

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 2]
```diff
@@ -5,79 +5,101 @@
...
+                            </label>
                         </div>
                     </ng-container>
                 </div>
                 <div class="form-group">
-                    <label for="FirstName" aria-required="true" class="required">First name</label>
-                    <input id="firstName" class="form-control" type="text" required placeholder="First Name"
-                        formControlName="firstName">
+                    <label>
+                        <span class="label-content required" aria-required="true">First name</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]: `aria-required` not needed when we tag input as required
```suggestion
                        <span class="label-content required">First name</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 3]
```diff
@@ -5,79 +5,101 @@
...
+                        <input class="form-control" type="text" required placeholder="First Name"
+                            formControlName="firstName">
+                    </label>
                 </div>
                 <div class="form-group">
-                    <label for="LastName" aria-required="true" class="required">Last name</label>
-                    <input id="lastName" type="text" class="form-control" required placeholder="Last Name"
-                        formControlName="lastName">
+                    <label>
+                        <span class="label-content required" aria-required="true">Last name</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                        <span class="label-content required">Last name</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 4]
```diff
@@ -5,79 +5,101 @@
...
+                        <input type="text" class="form-control" required placeholder="Last Name"
+                            formControlName="lastName">
+                    </label>
                 </div>
                 <div class="form-group">
-                    <label for="Address1" aria-required="true" class="required">Address 1</label>
-                    <input id="address1" type="text" class="form-control" required placeholder="Street Address"
-                        formControlName="line1">
+                    <label>
+                        <span class="label-content required" aria-required="true">Address 1</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                        <span class="label-content required">Address 1</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 5]
```diff
@@ -5,79 +5,101 @@
...
                 </div>
                 <div class="form-group">
                     <ng-container *ngIf="countries$ | async as countries">
                         <div *ngIf="countries.length !== 0">
-                            <label for="Country" aria-required="true" class="required">Country</label>
-                            <ng-select [searchable]="false" [clearable]="false" [items]="countries" bindLabel="name"
-                                bindValue="isocode" placeholder="Select One..." (change)="countrySelected($event)">
-                            </ng-select>
+                            <label>
+                                <span class="label-content required" aria-required="true">Country</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                                <span class="label-content required">Country</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 6]
```diff
@@ -5,79 +5,101 @@
...
+                            </label>
                         </div>
                     </ng-container>
                 </div>
                 <div class="row">
                     <div class="form-group col-md-6">
-                        <label for="City" aria-required="true" class="required">City</label>
-                        <input id="city" type="text" class="form-control" required placeholder="City" formControlName="town">
+                        <label>
+                            <span class="label-content required" aria-required="true">City</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                            <span class="label-content required">City</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 7]
```diff
@@ -5,79 +5,101 @@
...
+                    </label>
                 </div>
                 <div class="form-group">
                     <ng-container *ngIf="countries$ | async as countries">
                         <div *ngIf="countries.length !== 0">
-                            <label for="Country" aria-required="true" class="required">Country</label>
-                            <ng-select [searchable]="false" [clearable]="false" [items]="countries" bindLabel="name"
-                                bindValue="isocode" placeholder="Select One..." (change)="countrySelected($event)">
-                            </ng-select>
+                            <label>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                            <label aria-required="true">
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 8]
```diff
@@ -5,79 +5,101 @@
...
-                                        bindValue="region" placeholder="Select One..." (change)="regionSelected($event)">
-                                    </ng-select>
-                                </ng-container>
-                                <ng-container *ngIf="!regions[0].name">
-                                    <ng-select [searchable]="false" [clearable]="false" [items]="regions" bindLabel="isocode"
-                                        bindValue="region" placeholder="Select One..." (change)="regionSelected($event)">
-                                    </ng-select>
-                                </ng-container>
+                                <label>
+                                    <span class="label-content required" aria-required="true">State</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                                    <span class="label-content required">State</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 9]
```diff
@@ -5,79 +5,101 @@
...
-                                    <ng-select [searchable]="false" [clearable]="false" [items]="regions" bindLabel="name"
-                                        bindValue="region" placeholder="Select One..." (change)="regionSelected($event)">
-                                    </ng-select>
-                                </ng-container>
-                                <ng-container *ngIf="!regions[0].name">
-                                    <ng-select [searchable]="false" [clearable]="false" [items]="regions" bindLabel="isocode"
-                                        bindValue="region" placeholder="Select One..." (change)="regionSelected($event)">
-                                    </ng-select>
-                                </ng-container>
+                                <label>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                                <label aria-required="true">
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 10]
```diff
@@ -5,79 +5,101 @@
...
+                                </label>
                             </div>
                         </ng-container>
                     </div>
                     <div class="form-group col-md-6">
-                        <label for="ZipCode" aria-required="true" class="required">Zip Code</label>
-                        <input id="zipCode" type="text" class="form-control" required placeholder="Postal Code/Zip"
-                            formControlName="postalCode">
+                        <label>
+                            <span class="label-content required" aria-required="true">Zip code</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                            <span class="label-content required">Zip code</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/payment-method/payment-form/payment-form.component.html [hunk 11]
```diff
@@ -6,50 +6,67 @@
                 <div class="form-group">
                     <ng-container *ngIf="cardTypes$ | async as cardTypes">
                         <div *ngIf="cardTypes.length !== 0">
-                            <label for="CardType" aria-required="true">Payment Type</label>
-                            <ng-select [searchable]="false" [clearable]="false" [items]="cardTypes" bindLabel="name"
-                                bindValue="code" placeholder="Select One..." (change)="paymentSelected($event)">
-                            </ng-select>
+                            <label>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/payment-method/payment-form/payment-form.component.html]:
```suggestion
                            <label aria-required="true">
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 12]
```diff
@@ -5,79 +5,101 @@
                 <div class="form-group">
                     <ng-container *ngIf="titles$ | async as titles">
                         <div *ngIf="titles.length !== 0">
-                            <label for="Title" aria-required="true" class="required">Title</label>
-                            <ng-select [searchable]="false" [clearable]="false" [items]="titles" bindLabel="name"
-                                bindValue="code" placeholder="Select Title..." (change)="titleSelected($event)">
-                            </ng-select>
+                            <label>
+                                <span class="label-content required" aria-required="true">Title</span>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                                <span class="label-content required">Title</span>
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html [hunk 13]
```diff
@@ -5,79 +5,101 @@
                 <div class="form-group">
                     <ng-container *ngIf="titles$ | async as titles">
                         <div *ngIf="titles.length !== 0">
-                            <label for="Title" aria-required="true" class="required">Title</label>
-                            <ng-select [searchable]="false" [clearable]="false" [items]="titles" bindLabel="name"
-                                bindValue="code" placeholder="Select Title..." (change)="titleSelected($event)">
-                            </ng-select>
+                            <label>
```
Comment (marlass) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/shipping-address/address-form/address-form.component.html]:
```suggestion
                            <label aria-required="true">
```

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/payment-method/payment-form/payment-form.component.html (hunk 1 above)
Comment (Matejk00) [reply projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/payment-method/payment-form/payment-form.component.html]: I deleted attribute from span.

**Pull request merged**